go/enclave/rpc/GetTransactionReceipt.go

```diff
@@ -57,3 +76,108 @@ func GetTransactionReceiptExecute(builder *CallBuilder[gethcommon.Hash, map[stri
 	builder.ReturnValue = &r
 	return nil
 }
+
+func fetchFromCache(ctx context.Context, storage storage.Storage, cacheService *storage.CacheService, txHash gethcommon.Hash, requester *gethcommon.Address) (map[string]interface{}, error) {
+	rec, _ := cacheService.ReadReceipt(ctx, txHash)
+	if rec == nil {
+		return nil, nil
+	}
+
+	// receipt found in cache
+	// for simplicity only the tx sender will access the cache
+	// check whether the requester is the sender
+	if rec.From != requester {
+		return nil, nil
+	}
+
+	logs := rec.Receipt.Logs
+	// filter out the logs that the sender can't read
+	// doesn't apply to value transfers (when to=nil)
+	if rec.To != nil && *rec.To != (gethcommon.Address{}) {
+		ctr, err := storage.ReadContract(ctx, *rec.To)
+		if err != nil {
+			return nil, fmt.Errorf("could not read contract in eth_getTransactionReceipt request. Cause: %w", err)
+		}
+		logs, err = filterLogs(ctx, storage, rec.Receipt.Logs, ctr, requester)
+		if err != nil {
+			return nil, fmt.Errorf("could not filter cached logs in eth_getTransactionReceipt request. Cause: %w", err)
+		}
+	}
+	r := marshalReceipt(rec.Receipt, logs, rec.From, rec.To)
+	return r, nil
+}
+
+func filterLogs(ctx context.Context, storage storage.Storage, logs []*types.Log, ctr *enclavedb.Contract, requester *gethcommon.Address) ([]*types.Log, error) {
+	filtered := make([]*types.Log, 0)
+	for _, l := range logs {
+		canView, err := canViewLog(ctx, storage, ctr, l, requester)
+		if err != nil {
+			return nil, err
+		}
+		if canView {
+			filtered = append(filtered, l)
+		}
+	}
+	return filtered, nil
+}
+
+func canViewLog(ctx context.Context, storage storage.Storage, ctr *enclavedb.Contract, l *types.Log, requester *gethcommon.Address) (bool, error) {
+	eventSig := l.Topics[0]
+	eventType, err := storage.ReadEventType(ctx, ctr.Address, eventSig)
+	if err != nil {
+		return false, fmt.Errorf("could not read event type in eth_getTransactionReceipt request. Cause: %w", err)
+	}
+	// event visibility logic
+	canView := eventType.IsPublic() ||
+		(eventType.AutoPublic != nil && *eventType.AutoPublic) ||
+		(eventType.SenderCanView != nil && *eventType.SenderCanView) ||
```

Comment:
From the method signature it looks like this checks for a generic requester but actually you can only call this method when you've verified sender == requester. Worth clarifying somewhere with a comment or in the method name I think.

Reply:
I actually had a comment like that and removed it